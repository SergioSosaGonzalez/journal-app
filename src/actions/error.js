import { types } from '../types/types';

export const setError = (text) => ({ type: types.uiSetError, payload: text });

export const removeError = () => ({
  type: types.uiRemoveError,
});
