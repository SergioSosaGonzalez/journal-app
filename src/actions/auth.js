import { types } from '../types/types';
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import Swal from 'sweetalert2';
import { notesLogout } from './notes';

export const startLoginWithEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch({ type: types.uiStartLoading, payload: {} });
    return firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch({ type: types.uiFinishLoading, payload: {} });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
        dispatch({ type: types.uiFinishLoading, payload: {} });
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    console.log('hola mudno');
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};

export const logout = () => {
  return { type: types.logout };
};

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();
    dispatch(logout());
    dispatch(notesLogout());
  };
};
