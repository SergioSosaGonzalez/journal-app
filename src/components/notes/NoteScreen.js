import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { activeNote, startDeleting } from '../../actions/notes';
import { useForm } from '../../hooks/useForm';
import { NoteAppbar } from './NoteAppbar';

const NoteScreen = () => {
  const dispatch = useDispatch();
  const { active: note } = useSelector((state) => state.notes);
  const [formValues, handleInputChange, reset] = useForm(note);
  const { body, title, url } = formValues;

  const activeId = useRef(note.id);

  useEffect(() => {
    if (note.id !== activeId.current) {
      reset(note);
      activeId.current = note.id;
    }
  }, [note, reset]);

  useEffect(() => {
    dispatch(activeNote(note.id, { ...formValues }));
  }, [formValues, dispatch]);

  const handleDelete = () => {
    dispatch(startDeleting(note.id));
  };
  return (
    <div className="notes__main-content">
      <NoteAppbar />
      <div className="notes__content">
        <form>
          <input
            type="text"
            placeholder="Some awesome title"
            className="notes__title-input"
            value={title}
            name="title"
            onChange={handleInputChange}
          />
          <textarea
            placeholder="What happen today"
            className="notes__text-area"
            value={body}
            name="body"
            onChange={handleInputChange}></textarea>
          <div className="notes__image">
            {note.url && <img src={note.url} alt="img" />}
          </div>
        </form>
      </div>

      <button className="btn btn-danger" onClick={handleDelete}>
        Delete
      </button>
    </div>
  );
};
export default NoteScreen;
