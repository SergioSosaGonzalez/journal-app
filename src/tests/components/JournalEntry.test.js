import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import JournalEntry from '../../components/journal/JournalEntry';
import { activeNote } from '../../actions/notes';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {
    uid: '1',
    name: 'Sergio',
  },
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: null,
    notes: [],
  },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const nota = {
  id: 10,
  date: 0,
  title: 'hola',
  body: 'Mundo',
  url: 'asdasd',
};

describe('Pruebas del JournalEntry', () => {
  const wrapper = mount(
    <Provider store={store}>
      <JournalEntry {...nota} />
    </Provider>
  );
  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de activar la nota', () => {
    wrapper.find('.journal__entry').prop('onClick')();
    expect(store.dispatch).toHaveBeenCalled();

    expect(store.dispatch).toHaveBeenCalledWith(
      activeNote(nota.id, { ...nota })
    );
  });
});
