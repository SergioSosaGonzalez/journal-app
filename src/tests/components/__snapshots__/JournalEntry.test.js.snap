// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del JournalEntry Debe mostrarse correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(asdasd)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      hola
    </p>
    <p
      className="journal__entry-content"
    >
      Mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
